client/my-sites/plans-v2/plan-renewal-notice/index.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent, ReactElement } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { OFFER_RESET_SUPPORT_PAGE, PLAN_RENEWAL_RECOMMENDATION } from '../constants';
+import { isEligibleForRenewalAtOldRate, slugToSelectorProduct } from '../utils';
+import ExternalLink from 'components/external-link';
+import { useLocalizedMoment } from 'components/localized-moment';
+
+/**
+ * Type dependencies
+ */
+
+import type { Purchase } from 'lib/purchases/types';
+
+type Props = {
+	purchase: Purchase;
+};
+
+const PlanRenewalNotice: FunctionComponent< Props > = ( { purchase } ) => {
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	let notice;
+
+	if ( isEligibleForRenewalAtOldRate( purchase, moment ) ) {
+		notice = translate(
+			"We're updating our plans to include new features and improved functionality. " +
+				'As a thank you for being a loyal Jetpack subscriber, you can renew your current plan one time – your choice, for one month or one year. ' +
+				'After that, you will have to move to a new plan.'
+		);
+	} else {
+		notice = translate(
+			"We're sorry, your current plan is no longer available. Please select a new option."
+		);
+
+		const { productSlug: currentSlug } = purchase;
+		const recommendedSlug = PLAN_RENEWAL_RECOMMENDATION[ currentSlug ];
+
+		if ( recommendedSlug ) {
+			notice = (
+				<>
+					{ notice }&nbsp;
+					{ translate(
+						'Based on your usage of the %(currentPlan)s plan, we recommend {{recommendedPlan/}}.',
+						{
+							comment:
+								'`currentPlan` refers to the current deprecated plan. `recommendedPlan` is the plan we recommend instead.',
+							args: {
+								currentPlan: slugToSelectorProduct( currentSlug )?.displayName,
+							},
+							components: {
+								recommendedPlan: slugToSelectorProduct( recommendedSlug )
```

Comment:
Non-blocker: would be good to put this to a const, as this function does several array iterations and function calls that can be re-used.

Reply:
I'm not sure I follow. Are you talking about what's returned by `slugToSelectorProduct`? Function's called twice but with 2 different arguments.

Reply:
Wow, I need my coffee apparently. 😅